src/user-intended-play-control.js

```diff
@@ -0,0 +1,104 @@
+function wikiaJWPlayerUserIntendedPlayControl(isInitiallyUserIntendedPlay, playerInstance, tracker, willAutoplay) {
+	var isUserIntendedPlay = null;
+	var userIntendedPlayReason = null;
+	var wasPausedByUserInteraction = false;
+	var reasonsForUserIntendedPlay = {
+		unpausing: 'unpausing',
+		fullScreen: 'fullscreen',
+		unmuting: 'unmuting',
+		videoThumbnailInsidePlayer: 'videothumbnailinsideplayer',
+		clickToPlay: 'clicktoplay',
+		playerInitializedByUserInteraction: 'playerinitializedbyuserinteraction'
+	};
+
+	function setUserIntendedPlay(isUserIntended, userIntendedReason, immediate) {
+		if (isUserIntendedPlay === isUserIntended) {
+			return;
+		}
+
+		isUserIntendedPlay = isUserIntended;
+
+		if (isUserIntended) {
+			userIntendedPlayReason = userIntendedReason;
+		} else {
+			userIntendedReason = null;
+		}
+
+		if (typeof tracker.setCustomDimension !== 'function') {
+			return;
+		}
+
+		if (immediate) {
+			tracker.setCustomDimension(39, isUserIntended ? 'user-intended' : 'not-user-intended');
+		} else {
+			// Related video impression happens just before the potential subsequent video play
+			playerInstance.on('relatedVideoImpression', function () {
+				tracker.setCustomDimension(39, isUserIntended ? 'user-intended' : 'not-user-intended');
+			});
+		}
+	}
+
+	function onPause(data) {
+		if (data.pauseReason === 'interaction') {
+			wasPausedByUserInteraction = true;
+		}
+	}
+
+	function onPlay(data) {
+		if (wasPausedByUserInteraction) {
+			setUserIntendedPlay(true, reasonsForUserIntendedPlay.unpausing);
+		}
+	}
+
+	function onFullScreen() {
+		setUserIntendedPlay(true, reasonsForUserIntendedPlay.fullScreen);
+	}
+
+	function onUnmute() {
+		setUserIntendedPlay(true, reasonsForUserIntendedPlay.unmuting);
+	}
+
+	function onMute() {
+		if (isUserIntendedPlay && userIntendedPlayReason === reasonsForUserIntendedPlay.unmuting) {
```

Comment:
As we talked offline please get rid of `reasonsForUserIntendedPlay` and add flag which determines if the reason of user intended play was unmuting. Or even we can get rid of this `if` completely. Ask product.